### PR TITLE
[SYCL][Doc] Improve ESIMD docs rendering in doxygen

### PR DESCRIPTION
**Pull request by alexbatashev**

### CODE REVIEW

**Review by kbobrovs — APPROVED** [review]
Thanks a lot!

File: sycl/include/CL/sycl/INTEL/esimd/esimd_memory.hpp [hunk 1]
```diff
@@ -689,27 +710,21 @@ ESIMD_NODEBUG ESIMD_INLINE
...
-// @param x X-coordinate of the left upper rectangle corner in BYTES.
-//
-// @param y Y-coordinate of the left upper rectangle corner in ROWS.
-//
-// @return the linearized 2D block data read from surface.
-//
+/// Media block load.
+///
+/// \tparam T is the element data type.
+/// \tparam m is the hight of the 2D block.
```
Comment (sergey-semenov) [inline sycl/include/CL/sycl/INTEL/esimd/esimd_memory.hpp:718]:
```suggestion
/// \tparam m is the height of the 2D block.
```

File: sycl/include/CL/sycl/INTEL/esimd/esimd_memory.hpp [hunk 2]
```diff
@@ -739,27 +754,19 @@ media_block_load(AccessorTy acc, unsigned x, unsigned y) {
...
-//
-// @param y Y-coordinate of the left upper rectangle corner in ROWS.
-//
-// @param vals the linearized 2D block data to be written to surface.
-//
-/// Media block store
+/// Media block store.
+///
+/// \tparam T is the element data type.
+/// \tparam m is the hight of the 2D block.
```
Comment (sergey-semenov) [inline sycl/include/CL/sycl/INTEL/esimd/esimd_memory.hpp:760]:
```suggestion
/// \tparam m is the height of the 2D block.
```

**Review by sergey-semenov — APPROVED** [review]
LGTM, some minor typos carried over from the old comments, feel free to apply separately.

**Review by sergey-semenov — APPROVED** [review]

**Pull request merged**